great_expectations/data_context/data_context.py

```diff
@@ -1268,6 +1268,7 @@ def profile_datasource(self, datasource_name, generator_name=None, profiler=Basi
                     data_asset_name=NormalizedDataAssetName(datasource_name, generator_name, name),
                     expectation_suite_name=profiler.__name__
                 )
+                batch.discard_subset_failing_expectations = False
```

Comment:
Unfortunately, this parameter is probably confusingly named: it controls what happens (in Pandas) when a new dataset object is created from a different one (but does not control what happens when one calls `get_expectation_suite()`